src/utils/pusher.tsx

```diff
@@ -4,62 +4,74 @@
  * This defines a Pusher client and channel connection as a vanilla Zustand store.
  */
 import Pusher, { Channel, PresenceChannel } from "pusher-js";
-import reactZustandCreate from "zustand";
 import vanillaCreate, { StoreApi } from "zustand/vanilla";
 
 const pusher_key = process.env.NEXT_PUBLIC_PUSHER_APP_KEY!;
+const pusher_server_host = process.env.NEXT_PUBLIC_PUSHER_SERVER_HOST!;
+const pusher_server_port = parseInt(process.env.NEXT_PUBLIC_PUSHER_SERVER_PORT!, 10);
+const pusher_server_tls = process.env.NEXT_PUBLIC_PUSHER_SERVER_TLS === 'true';
 
 interface PusherZustandStore {
   pusherClient: Pusher;
   channel: Channel;
   presenceChannel: PresenceChannel;
-
   members: { [key: string]: any };
 }
 
-// self-contained Pusher state machine that manages:
-// - Pusher Client
-// - Connected/subscribed channels
-// - "Current members" via presence
-const createPusherStore = (slug: string, publishPresence?: boolean) => {
-  const randomUserId = `random-user-id:${Math.random().toFixed(7)}`;
-  const pusherClient = new Pusher(pusher_key, {
-    authEndpoint: "/api/pusher/auth-channel",
-    auth: {
-      headers: { user_id: randomUserId },
-    },
-    cluster: "us3",
-  });
+const usePusherStore = (slug: string) => {
+  let pusherClient : Pusher;
+  if (Pusher.instances.length) {
+    pusherClient = Pusher.instances[0] as Pusher;
+    pusherClient.connect();
+  } else {
+    const randomUserId = `random-user-id:${Math.random().toFixed(7)}`;
+    pusherClient = new Pusher(pusher_key, {
+      wsHost: pusher_server_host,
+      wsPort: pusher_server_port,
```

Comment:
I was having a lot of issues connecting to railway when passing a "port" - curious if you ran into those as well 🤔

Reply:
I didn't have any issues with the port. I think when you deploy on railway use 443 since it's hosted on https and if you run locally it's 6001 like I put in the `.env.example`. Could definitely add a note to the readme about this

Reply:
Yeah add that note if you don't mind as this would have bit me